src/Engine/ProtoCore/Utils/CoreUtils.cs

```diff
@@ -796,6 +797,38 @@ public static AssociativeNode CreateNodeFromString(string name)
             return newIdentList;
         }
 
+        public static AssociativeNode CreateNodeByCombiningIdentifiers(IList<AssociativeNode> nodeList)
```

Comment:
Added a new utility API to create a new `IdentifierListNode` from an array of input nodes in order.